src/scenes/WeatherHourlyList/WeatherHourlyList.js

```diff
@@ -1,59 +1,66 @@
-        hours.sort((a, b) => a - b)
-        // return hours.map((hour) => {
-        //     const foundWeather = weatherList.find((weather) => new Date(weather.created).getHours() == hour)
-        //     foundWeather.ihourly = hour
-        //     return foundWeather
-        // })
-        return hours.map((hour) => ({...weatherList.find((weather) => new Date(weather.created).getHours() == hour), hour }))
-    }, [weatherList])
+  const { match: { params: { locationId, year, month, day } } } = props
+  const date = year + month + day
+  const { data: weatherList, loading, error } = useFetchData(`${locationUri}/${locationId}/${date}`, [], [locationId, date])
+  const [filteredWeatherList, setFilteredWeatherList] = useState([])
 
-    return (
-        <React.Fragment>
-            <h3>{convertDateToDayName(`${year}/${month}/${day}`)}</h3>
-            <Table striped bordered hover>
-                <thead>
-                    <tr>
-                        <th>Hourly</th>
-                        <th>Weather State Name</th>
-                        <th>Min Temp</th>
-                        <th>Max Temp</th>
-                        <th>Humidity</th>
-                    </tr>
-                </thead>
-                <tbody>
-                    {sortedWeatherList.map((weather) => {
-                        return <WeatherHourly {...weather} />
-                    })}
-                </tbody>
-            </Table>
-            {error && <div className="text-danger">{error}</div>}
-            {loading && <div className="text-info">loading...</div>}
-        </React.Fragment >
-    )
-}
+  const sortedWeatherList = useMemo(() => {
+    const hours = [...new Set(weatherList.map((weather) => new Date(weather.created).getHours()))]
+    hours.sort((a, b) => a - b)
+    return hours.map((hour) => ({ ...weatherList.find((weather) => new Date(weather.created).getHours() === hour), hour }))
+  }, [weatherList])
 
-export default WeatherHourlyList;
\ No newline at end of file
+  useEffect(() => {
+    setFilteredWeatherList(sortedWeatherList)
+  }, [sortedWeatherList])
+
+  const handleSearch = (text) => {
+    const filterWeatherList = sortedWeatherList.filter((temp) =>
+      temp.weather_state_name.toLowerCase().includes(text) || (temp.humidity.toString() + '%').includes(text))
+    setFilteredWeatherList(filterWeatherList)
+  }
+  return (
+    <React.Fragment>
+      <WeatherStateSearchBar handleSearch={handleSearch} />
+      <h3>{convertDateToDayName(`${year}/${month}/${day}`)}</h3>
+      <Table striped bordered hover>
+        <thead>
+          <tr>
+            <th>Hourly</th>
+            <th>Weather State Name</th>
+            <th>Min Temp</th>
+            <th>Max Temp</th>
+            <th>Humidity</th>
+          </tr>
+        </thead>
+        <tbody>
+          {filteredWeatherList.map((weather) => {
+            // eslint-disable-next-line react/jsx-key
```

Comment:
Add jsx-key, chose a proper property for the key, maybe `hour`

Reply:
done